backend/curation/api/v1/curation/collections/common.py

```diff
@@ -198,6 +208,20 @@ def reshape_dataset_for_curation_api(dataset_version: DatasetVersion, use_canoni
     return ds
 
 
+def reshape_dataset_for_curation_api(dataset_version: DatasetVersion, use_canonical_url: bool, preview=False) -> dict:
+    dataset = reshape_dataset_for_curation_api_common(dataset_version, use_canonical_url, preview)
+    if not preview:
+        dataset["published_at"] = dataset_version.canonical_dataset.published_at
+        dataset["revised_at"] = dataset_version.canonical_dataset.revised_at
```

Comment:
I might be missing it, but are we setting the dataset_version.canonical_dataset revised_at anywhere? If we're not, I think this is why the test is failing with an unexpected None

Reply:
You are correct!